src/pages/reference/combining-multiple-sources.md

```diff
@@ -0,0 +1,489 @@
+}
+```
+
+## 1. Add the "Authors" Source
+
+The "Authors" Source is a gRPC API: [`authors-service`](https://github.com/charlypoly/graphql-mesh-docs-first-gateway/tree/master/packages/authors-service).
+
+We will use the `grpc` Handler with the `@graphql-mesh/grpc` package and configure in our [`.meshrc.yaml`](https://github.com/charlypoly/graphql-mesh-docs-first-gateway/tree/master/packages/multiple-sources/.meshrc.yaml) it as follows:
+
+```json
+{
+    "sources": [
+        {
+            "name": "Books",
+            "handler": {
+                "openapi": {
+                    "baseUrl": "http://localhost:3002/",
+                    "source": "../books-service/openapi3-definition.json"
+                }
+            }
+        },
+        {
+            "name": "Authors",
+            "handler": {
+                "grpc": {
+                    "endpoint": "localhost:3003",
+                    "protoFilePath": "../authors-service/proto/authors/v1/authors_service.proto"
+                }
+            }
+        }
+    ]
+}
+```
+
+Similar to the "Books" API configuration, we just need to provide the path to the service definition file (here, a Proto file) and the base URL.
+
+We will "clean" the corresponding "Authors" GraphQL Schema later, in the 3rd step, [Shaping the Unified Schema](#3-shaping-the-unified-schema).
+
+## 2. Add the "Stores" Source
+
+The "Stores" Source is a GraphQL API: [`stores-service`](https://github.com/charlypoly/graphql-mesh-docs-first-gateway/tree/master/packages/stores-service).
+
+We will use the `graphql` Handler with the `@graphql-mesh/graphql` package and configure it in our [`.meshrc.yaml`](https://github.com/charlypoly/graphql-mesh-docs-first-gateway/tree/master/packages/multiple-sources/.meshrc.yaml) it as follows:
+
+```json
+{
+    "sources": [
+        {
+            "name": "Books",
+            "handler": {
+                "openapi": {
+                    "baseUrl": "http://localhost:3002/",
+                    "source": "../books-service/openapi3-definition.json"
+                }
+            }
+        },
+        {
+            "name": "Authors",
+            "handler": {
+                "grpc": {
+                    "endpoint": "localhost:3003",
+                    "protoFilePath": "../authors-service/proto/authors/v1/authors_service.proto"
+                }
+            }
+        },
+        {
+            "name": "Stores",
+            "handler": {
+                "graphql": {
+                    "endpoint": "http://0.0.0.0:3004/graphql"
+                }
+            }
+        }
+    ]
+}
+```
+
+Since the "Stores" is a GraphQL API, Mesh can leverage introspection to get its schema with just the `endpoint` URL.
+
+## 3. Shaping the Unified Schema
+
+The following Mesh Gateway configuration:
+
+```json
+{
+    "sources": [
+        {
+            "name": "Books",
+            "handler": {
+                "openapi": {
+                    "baseUrl": "http://localhost:3002/",
```

Comment:
Can we replace these with placeholders or a dummy URL just for example purposes?

Reply:
thanks, @revanth0212. I have replaced the `localhost` paths with `<your_url>`